### ISSUE TITLE
11.0.0 Introduced false positives for `no-identical-title` for tests in separate suites in the same file

### ISSUE THREAD

**Issue by devm33** [lo1tuma/eslint-plugin-mocha#381]
Upgrading from `10.5.0` to `11.0.0`, getting newly false positives for `no-identical-title` when there are test names reused across separate suites that are in the same file (of course have `max-top-level-suites` off unfortunately 😏)

**Comment by e22m4u** [comment]
Same here

![Image](https://github.com/user-attachments/assets/11b17fa0-cb91-4580-9ef9-29121f7dc2bb)

**Comment by e22m4u** [comment]
Oh, God... It took me 3 hours to find workaround.
Do not `import {describe} from 'mocha';` to test files.

**Comment by lo1tuma** [comment]
@devm33 do you import mocha or do you have this issue when using globals?

@e22m4u Since v11 this plugin tries to correctly resolve the mocha functions, but it needs to know the [`interface`](https://mochajs.org/#interfaces) in advance. The default `interface` is `BDD` which assumes using the globals. So shadowing e.g. `describe` by importing it would lead the plugin to think `describe` is something different.
If you want to use mocha’s `exports` interface you should define `'mocha/interface': 'exports'` in you eslint config `settings`.

**Comment by devm33** [comment]
@lo1tuma using globals, not importing. We are using the `tdd` interface though and also TypeScript.

mocha options:

```js
    require: ['tsx'],
    ui: 'tdd',
```

eslint flat config for tests:

```js
import mochaPlugin from 'eslint-plugin-mocha';
```

```js
    {
        files: ['**/*.test.ts{,x}'],
        plugins: {mocha: mochaPlugin},
        rules: {
            ...mochaPlugin.configs.flat.recommended.rules,
            'mocha/max-top-level-suites': 'off',
        }
    }
```

**Comment by lo1tuma** [comment]
I can’t reproduce. I’ve added a test [here](https://github.com/lo1tuma/eslint-plugin-mocha/pull/388) but it passes.
Are you sure you are not using `import { suite, test } from 'mocha'` in your test file?

**Comment by devm33** [comment]
Ah! Apologies @lo1tuma you are correct! There were some `import {suite} from 'mocha';` that slipped in. Removing fixes the `no-identical-title` error. Thank you!
